spec/services/admin/npq_applications/applications_search_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Admin::NPQApplications::ApplicationsSearch, :with_default_schedules do
+  let(:search) { Admin::NPQApplications::ApplicationsSearch }
+
+  let!(:school) { create(:school) }
+  let!(:application_1) { create(:npq_application, school_urn: school.urn) }
+  let!(:application_2) { create(:npq_application, school_urn: school.urn) }
+  let!(:user) { application_1.user }
+
+  subject { described_class.new(query_string:) }
+
+  describe "#call" do
+    context "when partial email match" do
+      let(:query_string) { user.email.split("@").first }
+
+      it "returns the hit" do
+        expect(subject.call).to include(application_1)
+      end
+    end
+
+    context "when user#id match" do
+      let(:query_string) { user.id }
+
+      it "returns the hit" do
+        expect(subject.call).to include(application_1)
+      end
+    end
+
+    context "when partial name match" do
+      let(:query_string) { user.full_name[0, 3] }
+
+      it "returns the hit" do
+        expect(subject.call).to include(application_1)
+      end
```

Comment:
Perhaps worth expecting the non-hit to not be returned?

Reply:
Yeah thats good point! think just over looked that thanks..